create-custom-map-style/README.md

```diff
@@ -0,0 +1,990 @@
+					lat:  maputnikMap.getCenter().lat.toFixed(4),
+					zoom:  maputnikMap.getZoom().toFixed(2),
+				});
+			});
+			return () =>  maputnikMap.remove();
+		}, []);
+	})
+	...
+	
+	// And in your return statement below change
+	// <div className="maputnik-map" />
+	// to 
+	
+	<div ref={maputnik} className="maputnik-map" />
+	```
+
+6. You may notice that we're calling a `transformRequest` function in the constructor for the map. We'll need to add this now. Create a new file in the `src` directory called `transformRequest.js` and paste the following:
+
+	```
+	import { Signer } from  '@aws-amplify/core';
+
+	export  default (credentials) => (url, resourceType) => {
+		if (resourceType === "Style" && !url.includes("://")) {
+			url = `https://maps.geo.us-west-2.amazonaws.com/maps/v0/maps/${url}/style-descriptor`;
+		}
+		if (url.includes("amazonaws.com")) {
+			return {
+				url:  getSignedUrl(url, credentials),
+			};
+		}
+		return {
+			url
+		};
+	}
+
+	export  const  getSignedUrl = (url, credentials) => {
+		return  Signer.signUrl(url, {
+			access_key:  credentials.accessKeyId,
+			secret_key:  credentials.secretAccessKey,
+			session_token:  credentials.sessionToken,
+		});
+	}
+	```
+
+7. Be sure to import the function in your `SamplePage` component
+8. You should be able to go back to your browser now and see the base map style show up inside your `maputnik` div
+
+	![Esri Streets Base Map](media/BaseMap.png)
+	*Esri Streets Base Map*
+
+
+## Styling a Custom Map with Maputnik
+1. Maputnik is an open source visual editor for the [Mapbox Style Specification](https://www.mapbox.com/mapbox-gl-js/style-spec). At the time of writing this editor can be either used in the browser or downloaded as a binary and can run locally. We are going to use the latter option.
+2. Go to https://github.com/maputnik/editor/wiki/Maputnik-CLI and follow the instructions for installation or optionally from your terminal:
+	```
+	wget https://github.com/maputnik/editor/releases/download/latest/maputnik-darwin.zip
+	unzip maputnik-darwin.zip
+	chmod 755 maputnik
+	```
+3. Start up Maputnik 
+	```
+	./maputnik
+	```
+	And open your browser to http://localhost:8000/#12.44/40.7356/-74.0541 
+	
+4.  It would be best to start with a clean canvas. Click "Open" in the top navigation bar, then scroll down to the bottom of "Gallery Styles" and selected "Empty Style". 
+
+	![Open](media/MaputnikOpen.png)
+	*Open A Style*
+
+	![Empty](media/MaputnikEmpty.png)
+	*Empty Canvas*
+
+5. Next we need to add the Amazon Location Service API as a data source. However, in order to do this we need to use a local proxy that can handle SigV4 (TL;DR a way to provide your AWS credentials with http requests). For this we will use [tessera](https://github.com/mojodna/tessera). From a new terminal (the `maputnik` process will be occupying the other terminal) run the following: 
+	```
+	nvm use 12
+	npm i -g tessera tilejson tilelive-xray @mapbox/tilelive @mapbox/tilejson
```

Comment:
Do you need all of these?

Reply:
I'm not sure tbh... will have a second look and remove any we don't need

Reply:
I removed this whole section now so we're only using tessera and tilelive-aws to serve tiles now